Add functionality to sort & filter `ard_stack_hierarchical()` results

edelarua opened the pull request:
**What changes are proposed in this pull request?**
* Added functions `sort_ard_hierarchical()` and `filter_ard_hierarchical()` to sort & filter ARDs created using `ard_stack_hierarchical()` and `ard_stack_hierarchical_count()`. (#301)

Closes #301

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`
- [ ] Request a reviewer

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

_Optional Reverse Dependency Checks_:

Install `checked` with `pak::pak("Genentech/checked")` or `pak::pak("checked")`

```shell
# Check dev versions of `cardx`, `gtsummary`, and `tfrmt` which are in the `ddsjoberg` R Universe
Rscript -e "options(checked.check_envvars = c(NOT_CRAN = TRUE)); checked::check_rev_deps(path = '.', n = parallel::detectCores() - 2L, repos = c('https://ddsjoberg.r-universe.dev', 'https://cloud.r-project.org'))"

# Check CRAN reverse dependencies but run tests skipped on CRAN
Rscript -e "options(checked.check_envvars = c(NOT_CRAN = TRUE)); checked::check_rev_deps(path = '.', n = parallel::detectCores() - 2, repos = 'https://cloud.r-project.org')"

# Check CRAN reverse dependencies in a CRAN-like environment
Rscript -e "options(checked.check_envvars = c(NOT_CRAN = FALSE), checked.check_build_args = '--as-cran'); checked::check_rev_deps(path = '.', n = parallel::detectCores() - 2, repos = 'https://cloud.r-project.org')"
```

ddsjoberg commented:
Hi @edelarua , is this ready for review? Thanks!

edelarua commented:
> Hi @edelarua , is this ready for review? Thanks!

Hi @ddsjoberg, yes I think this should be ready now! I've also updated the corresponding PR in gtsummary (https://github.com/ddsjoberg/gtsummary/pull/2097)  to work with the updated version

ddsjoberg commented on R/ard_filter.R:
```diff
@@ -0,0 +1,109 @@
+#' @seealso [ard_sort()]
+#' @name ard_filter
+#'
+#' @examplesIf (identical(Sys.getenv("NOT_CRAN"), "true") || identical(Sys.getenv("IN_PKGDOWN"), "true"))
```
I added some additional examples with text. Can you confirm what I wrote is correct?

edelarua commented:
@ddsjoberg this is ready for another review!

I added the `keep_empty_summary` parameter to `filter_ard_hierarchical()` - if you can think of a better name for the parameter please share! I currently have it so that if all inner rows are filtered out and `keep_empty_summary = FALSE`, the overall rows (if `over_variables=TRUE`) is kept as well as the header info rows. Does that make sense?

edelarua commented:
> > @ddsjoberg this is ready for another review!
> > I added the `keep_empty_summary` parameter to `filter_ard_hierarchical()` - if you can think of a better name for the parameter please share! I currently have it so that if all inner rows are filtered out and `keep_empty_summary = FALSE`, the overall rows (if `over_variables=TRUE`) is kept as well as the header info rows. Does that make sense?
> 
> Thank you @edelarua !! I changed `keep_empty_summary` to `keep_empty` just to make it shorter. I think it's ok?
> 
> Can you add a note to the documentation about how the `over_variables=TRUE` results are handled?

`keep_empty` sounds good! I have added a note about `over_variables`.

ddsjoberg reviewed (approved):
@edelarua ! Thank you so so so much for this functionality! This is awesome!
